src/compiler/utilities.ts

```diff
@@ -3086,6 +3086,14 @@ export function getImmediatelyInvokedFunctionExpression(func: Node): CallExpress
     }
 }
 
+/** @internal */
+export function isAwaited(node: Node): boolean {
+    while(node.kind === SyntaxKind.ParenthesizedExpression) {

```

Comment:
I think you should use the `walkUpOuterExpressions` function from `src/compiler/factory/utilities.ts` instead.